contracts/zero-ex/tests/Multiplex.t.sol

```diff
@@ -0,0 +1,1453 @@
+            "uint256,"
+            "uint256"
+            ")"
+        );
+
+    struct LogDescription {
+        bytes32[] topics;
+        bytes data;
+        bytes dataMask;
+    }
+
+    DeployZeroEx.ZeroExDeployed private zeroExDeployed;
+    IFlashWallet private flashWallet;
+    IERC20TokenV06 private shib;
+    IERC20TokenV06 private dai;
+    IERC20TokenV06 private zrx;
+    IEtherTokenV06 private weth;
+
+    TestUniswapV2Factory private sushiFactory;
+    TestUniswapV2Factory private uniV2Factory;
+    TestUniswapV3Factory private uniV3Factory;
+    TestLiquidityProvider private liquidityProvider;
+    uint256 private transformerNonce;
+
+    address private signerAddress;
+    uint256 private signerKey;
+
+    modifier explain(string memory testName) {
+        log_string(testName);
+        vm.recordLogs();
+        _;
+    }
+
+    function infiniteApprovals() private {
+        shib.approve(address(zeroExDeployed.zeroEx), MAX_UINT256);
+        dai.approve(address(zeroExDeployed.zeroEx), MAX_UINT256);
+        zrx.approve(address(zeroExDeployed.zeroEx), MAX_UINT256);
+        weth.approve(address(zeroExDeployed.zeroEx), MAX_UINT256);
+    }
+
+    function setUp() public {
+        // TODO signer utilities shouldn't be in ForkUtils
+        (signerAddress, signerKey) = getSigner();
+
+        sushiFactory = new TestUniswapV2Factory();
+        uniV2Factory = new TestUniswapV2Factory();
+        uniV3Factory = new TestUniswapV3Factory();
+        liquidityProvider = new TestLiquidityProvider();
+
+        zeroExDeployed = new DeployZeroEx(
+            DeployZeroEx.ZeroExDeployConfiguration({
+                uniswapFactory: address(uniV2Factory),
+                sushiswapFactory: address(sushiFactory),
+                uniswapV3Factory: address(uniV3Factory),
+                uniswapPairInitCodeHash: uniV2Factory.POOL_INIT_CODE_HASH(),
+                sushiswapPairInitCodeHash: sushiFactory.POOL_INIT_CODE_HASH(),
+                uniswapV3PoolInitCodeHash: uniV3Factory.POOL_INIT_CODE_HASH(),
+                logDeployed: false
+            })
+        ).deployZeroEx();
+
+        transformerNonce = zeroExDeployed.transformerDeployer.nonce();
+        vm.prank(zeroExDeployed.transformerDeployer.authorities(0));
+        zeroExDeployed.transformerDeployer.deploy(type(TestMintTokenERC20Transformer).creationCode);
+
+        flashWallet = zeroExDeployed.zeroEx.getTransformWallet();
+
+        shib = IERC20TokenV06(address(new TestMintableERC20Token()));
+        dai = IERC20TokenV06(address(new TestMintableERC20Token()));
+        zrx = IERC20TokenV06(address(new TestMintableERC20Token()));
+        weth = zeroExDeployed.weth;
+
+        infiniteApprovals();
+        vm.startPrank(signerAddress);
+        infiniteApprovals();
+        vm.stopPrank();
+
+        vm.deal(address(this), 10e18);
+    }
+
+    // TODO refactor some of these utility functions out into helper contract
+
+    function makeTestRfqOrder(
```

Comment:
These `make.*{Order,Subcall}` functions ought to be in a library or at least in a separate contract.

Reply:
Yep I agree, hence the `TODO` immediately above. Just while this is WIP I have been dumping everything inside this contract. Probably a library in the same file would've been a little cleaner, the next version will at least have that.